vignettes/compatibility.Rmd

```diff
@@ -109,7 +109,7 @@ rle(1:10)
 Alternatively you could convert it to the most natural representation using S7:
 
 ```{r}
-rle <- new_class("rle", properties = list(
+new_rle <- new_class("rle", properties = list(
```

Comment:
Maybe `rle2`? I think calling it `new_` is a bit confusing because the while the object is constructor, it's also the class.

Reply:
I agree, `rle2` would be clearer than `new_rle`.

Reply:
There's a few other instances north of this line where the name `new_rle` is used that would have to be updated as well for the examples to work as the original author intended. I'll add that change. I agree that the examples are in this section of the vignette are a little difficult to reason about because the new S7 class is being asked to wear the hat of a formal class definition and a constructor for the "rle" class.

Reply:
Thanks for catching this! After looking more closely, I see that `new_rle()` is indeed the best choice here. The intent is for `rle()` to serve as a convenient constructor, and we show how to redefine the underlying `new_rle()` from the “old-style” S3 approach to the S7 approach (in two different ways) while keeping the user-facing `rle()` constructor unchanged.

Strictly speaking, it might make sense to replace the user-facing `rle()` with an S7 `rle` class that has a custom constructor, but that would increase the example’s size and complexity without adding much clarity, in my opinion. I’ve restored the previous `new_rle()` approach.